babelPlugin.js

```diff
@@ -30,8 +31,33 @@ module.exports = function nextBabelWrapper(nextConfig, pluginOptions) {
           // front matter is extracted and written out by the loader to .next/frontMatter/<filePathHashed>
           // here, we're calculating the path
           const currentPath = state.file.opts.filename
+
+          // Get absolute path to page
+          let generatedPath = path.resolve(path.dirname(currentPath), importPath);
+          
+          // The next steps serve to support placing pages under `src/pages`:
+          // 1. Checks whether `src/pages` exists
+          const usesSrc = fs.existsSync(path.join(nextConfig.dir, '/src/pages'));
+
+          // 2. Generate alternative path if `src/pages` exists, else use the one generated above
+          if(usesSrc) {
+            // 2.1 Construct new path 
+            
+            // >/dev/project/src/pages
+            const pathToPagesDirectory = nextConfig.dir + path.normalize('/src/pages');
+            
+            // >src/pages/docs/intro.mdx
+            const pathFromRoot = generatedPath.replace(nextConfig.dir, '');
+            
+            // >/docs/intro.mdx
+            const pathToPage = pathFromRoot.split('pages')[1];
+            
+            // >/dev/project/src/pages/docs/intro.mdx
+            generatedPath = pathToPagesDirectory + pathToPage;
+          }
+
```

Comment:
This section can be cut down by passing the `usesSrc` information from `index.js` into this plugin under `pluginOptions` rather than recreating it a second time.